src/lib/viewers/controls/color-picker/__tests__/ColorPickerControl-test.tsx

```diff
@@ -75,6 +75,19 @@ describe('ColorPickerControl', () => {
 
             expect(getColorPickerPalette(wrapper).hasClass('bp-is-open')).toBe(true);
         });
+
+        test('should call focus and stop propagation when mouse down', () => {
+            const mockEvent = {
+                currentTarget: {
+                    focus: jest.fn(),
+                },
+                preventDefault: jest.fn(),
+            };
+            (getToggleButton(getWrapper()).prop('onMouseDown') as Function)(mockEvent);
```

Comment:
I've been favoring `simulate` because it avoids the need to cast. I've seen the Jest folks say not to use it, but it's there and it works well. Up to you.

Reply:
Yeah I prefer using `simulate` too, and I tried it first but it doesn't work (I googled it why but it seems sometimes it happens and everyone just use props.onClick() etc. ). Then I have to use this fallback plan.